_posts/2018-08-24-How-to-debug-jest-tests.markdown

```diff
@@ -0,0 +1,30 @@
+---
+layout: epic
+title: "How To Debug Jest Tests"
+date: 2018-08-24
+author: anson
+categories: [Node.js, Jest, testing, tooling]
+---
+
+Hey there! My name is Anson and I work on the Platform team at Artsy. Recently, we faced an issue where a certain [Enzyme](https://github.com/airbnb/enzyme) test we wrote using mock tracking was failing, but we couldn't figure out why. Luckily, with some help from [Orta](/author/orta) and some clever thinking, we figured out what was going on.
+
+<!-- more -->
+
+We thought it was an issue with the mock testing library we had written. We tried to fix the problem by sprinkling `console.log` calls throughout the test, but it was still hard to figure out what was going on, especially without knowing how to peak into the properties of certain objects.
+
+Instead, [Orta](/author/orta) suggested we used the Chrome Node DevTools. Since the Enzyme test is run via `yarn jest`, yarn is acting as a frontend for running the Enzyme test with Node. This means that we can use the Chrome Node DevTools as a debugger to run the Enzyme test. This was super useful since the one thing we needed was to be able to peak inside certain objects to see what they looked like and how they were failing. It was a much faster, more methodical way to approach debugging this test. Here are the steps we took:
+
+- First, insert a new line in your test where you think it might be failing and type `debugger`. This will serve as a break point for the debugger to stop at.
+- Open up Chrome and type in the address bar : `chrome://inspect`
+- Click on "Open dedicated DevTools for Node"
+- In your terminal, instead of typing `yarn jest <path_to_test>`, type this:
+
+```bash
+node --inspect node_modules/.bin/jest --runInBand <path_to_test>
```

Comment:
It would be cool to maybe suggest an alias to this in `package.json`, something like `yarn test:debug`

Reply:
This got added 👍